### PR TITLE
Add preference menu for aug and sg

### PR DESCRIPTION
**Pull request by dcowen91**
Implements the feature for issue #25.

This makes it so that players can opt in to receiving scoped rifles on CT & T side. If they opt in, then there is a 25% chance that they will receive an sg/aug instead of an m4/ak. I'm not married to this percentage at all if you have any feedback on how to better incorporate this feature.

### CODE REVIEW

**Comment by splewis** [comment]
> If they opt in, then there is a 25% chance that they will receive an sg/aug instead of an m4/ak.

I think it'd be preferable to just have a generic rifle selection for both sides. CTs currently can pick silenced/unsilenced m4, the AUG should just be another entry.

**Comment by dcowen91** [comment]
> CTs currently can pick silenced/unsilenced m4, the AUG should just be another entry.

That makes sense. My only concern with letting every CT potentially have an AUG is that it might result in an unintended buff to CT sides, since the AUG makes certain angles a lot easier to hold. 

I'll go ahead and implement a generic rifle selection as you suggested and update this PR.

**Comment by dcowen91** [comment]
I've implemented the rifle selection for CT and T sides. Do note however that I have not been able to test these changes- I'm primarily a Windows dev and there are some compatibility issues I'm facing with running smbuilder inside Windows Subsystem for Linux. I'll keep banging my head against those issues, but wanted to update the PR in the meantime.

File: scripting/include/executes.inc [hunk 1]
```diff
@@ -27,6 +27,17 @@ enum StratType {
...
+enum CTRiflePref {
+  CTRiflePref_M4 = 0,
+  CTRiflePref_Silenced_M4 = 1,
+  CTRiflePref_Aug = 2,
+}
+
+enum TRiflePref {
+  TRiflePref_Ak = 0,
+  TRiflePref_Sg = 1,
+}
```
Comment (splewis) [inline scripting/include/executes.inc]: Move these somewhere else, anything in executes.inc is intended to be part of the public API.

File: scripting/executes.sp [hunk 2]
```diff
@@ -1106,10 +1118,12 @@ public void UpdateTeams() {
 
       if (StrEqual(g_LastItemPickup[client], "ak47")) {
         g_PlayerPrimary[client] = "weapon_ak47";
-      } else if (g_SilencedM4[client]) {
+      } else if (g_CTRifle[client] == CTRiflePref_M4) {
+        g_PlayerPrimary[client] = "weapon_m4a1";
```
Comment (splewis) [inline scripting/executes.sp]: I think this type of logic (same with the T side) should be encapsulated into another function `GivePlayerPrimaryRifle` or such. That way this logic doesn't need to be repeated so much.

File: scripting/executes.sp [hunk 3]
```diff
@@ -338,7 +337,6 @@ public void OnPluginStart() {
   }
 
   g_AllowAWPCookie = RegClientCookie("executes_awpchoice", "", CookieAccess_Private);
-  g_SilencedM4Cookie = RegClientCookie("executes_silencedm4", "", CookieAccess_Private);
   g_CZCTSideCookie = RegClientCookie("executes_cz_ct_side", "", CookieAccess_Private);
```
Comment (splewis) [inline scripting/executes.sp:355]: I'm not sure what the implications of using cookies without ever registering them with `RegClientCookie` is. I'd rather continue to register the new ones as well, unless there's some documentation in SM around what happens here.

**Comment by splewis** [comment]
> I've implemented the rifle selection for CT and T sides. Do note however that I have not been able to test these changes- I'm primarily a Windows dev and there are some compatibility issues I'm facing with running smbuilder inside Windows Subsystem for Linux. I'll keep banging my head against those issues, but wanted to update the PR in the meantime

Mostly looks good, a few notes.

You don't necessarily have to use smbuilder to compile this, you can compile it the 'standard way' for compiling sourcemod plugins (`spcomp executes.sp` or such). I mostly just use it to simplify packaging for servers & releases. (the travis.ci builds here are running the compilation too, so you *can* test it that way if you must :p)

**Comment by dcowen91** [comment]
Thanks for the feedback! I addressed the comments and was finally able to test it! :)

image

File: scripting/executes/loadout_common.sp [hunk 4]
```diff
@@ -42,7 +42,24 @@ public char RandomNade() {
   }
 }
 
-//
+public void GivePrimaryRifle(int client, int team) {
```
Comment (splewis) [inline scripting/executes/loadout_common.sp]: Rename this SetPrimaryRifle - GivePrimaryRifle makes me thing it's actually giving the weapon (like GivePlayerItem). Other than that everything looks good to me.

**Comment by splewis** [comment]
Thanks for testing. One last nitpick.

**Pull request merged**